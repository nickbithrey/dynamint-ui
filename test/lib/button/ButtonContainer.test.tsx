import * as React from 'react';
import { 
    shallow, 
    ShallowWrapper 
} from 'enzyme';
import {
    ButtonContainer,
    IButtonContainer
} from 'lib/button/ButtonContainer';
import {
    Button,
    IButton
} from 'lib/button/Button';

describe('<ButtonContainer />', () => {
    
    const props: IButtonContainer = {
        buttons: [
                  {key: 'button1', content: 'button1', onClick: jest.fn(), type: 'primary'},
                  {key: 'button2', content: 'button1', onClick: jest.fn(), type: 'default'}
              ]
    }
    
    let wrapper: ShallowWrapper;
    
    beforeEach(() => {
        wrapper = shallow(<ButtonContainer {...props} />);
    });
    
    it('renders a valid tag and matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('contains same number of buttons as supplied', () => {
        expect(wrapper.find('Button')).toHaveLength(props.buttons.length);
    });
    
    it('will not render anything if no buttons are supplied', () => {
        const emptyWrapper = shallow(<ButtonContainer {...{buttons: []}} />);
        expect(emptyWrapper.html()).toBeNull();
    });
    
    it('will throw error if buttons are not defined', () => {
        expect(() => shallow(<ButtonContainer {...{buttons: null}} />)).toThrowError();
        expect(() => shallow(<ButtonContainer {...{buttons: undefined}} />)).toThrowError();
    });
    
});