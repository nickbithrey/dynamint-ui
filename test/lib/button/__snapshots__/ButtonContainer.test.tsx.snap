// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonContainer /> renders a valid tag and matches snapshot 1`] = `
<Fragment>
  <Button
    content="button1"
    key="button1"
    onClick={[MockFunction]}
    type="primary"
  />
  <Button
    content="button1"
    key="button2"
    onClick={[MockFunction]}
    type="default"
  />
</Fragment>
`;
